src/shared/Core/Authentication/MicrosoftAuthentication.cs

```diff
@@ -279,20 +224,40 @@ private async Task<AuthenticationResult> GetAccessTokenSilentlyAsync(IPublicClie
                 appBuilder.WithLogging(OnMsalLogMessage, LogLevel.Verbose, enablePiiLogging, false);
             }
 
-            // If we have a parent window ID we should tell MSAL about it so it can parent any authentication dialogs
-            // correctly. We only support this on Windows right now as MSAL only supports embedded/dialogs on Windows.
-            if (PlatformUtils.IsWindows() && !string.IsNullOrWhiteSpace(Context.Settings.ParentWindowId) &&
-                int.TryParse(Context.Settings.ParentWindowId, out int hWndInt) && hWndInt > 0)
+            // On Windows we should set the parent window handle for the authentication dialogs
+            // so that they are displayed as a child of the correct window.
+            if (PlatformUtils.IsWindows())
             {
-                appBuilder.WithParentActivityOrWindow(() => new IntPtr(hWndInt));
+                // If we have a parent window ID then use that, otherwise use the hosting terminal window.
+                IntPtr parentHandle;
+                if (!string.IsNullOrWhiteSpace(Context.Settings.ParentWindowId) &&
+                    int.TryParse(Context.Settings.ParentWindowId, out int hWndInt) && hWndInt > 0)
+                {
+                    parentHandle = new IntPtr(hWndInt);
```

Comment:
How are you calculating the `hwndInt` ? For console apps it's a bit trickier - see https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/wam#parent-window-handles